### PR TITLE
Add CommandContext#shouldBroadcast and fix !code permission checks

### PR DESCRIPTION
**Pull request by Kaiepi**

### CODE REVIEW

File: server/chat-commands/info.js [hunk 1]
```diff
@@ -2401,16 +2401,17 @@ const commands = {
...
-		if (!this.canTalk()) return;
-		if (!this.runBroadcast(true, '!code')) return;
 		if (target.length >= 8192) return this.errorReply("Your code must be under 8192 characters long!");
 
 		const params = target.split('\n');
 		if (!params[0]) params.unshift();
 		if (!params[params.length - 1]) params.pop();
-		if (params.length === 1 && params[0].length < 80 && !target.includes('```') && this.message.startsWith('/')) {
-			return `\`\`\`${params[0]}\`\`\``;
+		if (params.length === 1 && params[0].length < 80 && !params[0].includes('```') && this.willBroadcast) {
```
Comment (Zarel) [inline server/chat-commands/info.js]: `this.willBroadcast` is always `false` here, isn't it?

Comment (Zarel) [reply server/chat-commands/info.js]: Also changing `target` to `params[0]` is pointless.

Comment (Kaiepi) [reply server/chat-commands/info.js]: `this.willBroadcast` is `false` with `/code`, but `true` with `!code`. `params[0]` may have fewer characters if there was a leading or trailing newline.

Comment (Zarel) [reply server/chat-commands/info.js]: I don't believe you.

Comment (Zarel) [reply server/chat-commands/info.js]: image

File: server/chat.ts [hunk 2]
```diff
@@ -733,8 +733,11 @@ export class CommandContext extends MessageContext {
 		}
 		return true;
 	}
+	get willBroadcast() {
```
Comment (Zarel) [inline server/chat.ts]: I hate getters. :[

Comment (Zarel) [reply server/chat.ts]: Just make it a function.

File: server/chat-commands/info.js (hunk 1 above)
Comment (Zarel) [reply server/chat-commands/info.js]: Oh, you introduced it in the PR. It shouldn't be a getter; getters are mostly banned.

File: server/chat.ts (hunk 2 above)
Comment (Zarel) [reply server/chat.ts]: Also I'd probably name it something like `shouldBroadcast()` for clarity?

Comment (Kaiepi) [reply server/chat.ts]: That sounds reasonable. What's wrong with getters though? I'm not attached to this being a getter or anything, just for reference in the future

**Comment by Kaiepi** [comment]
Rebased

File: server/chat.ts (hunk 2 above)
Comment (Zarel) [reply server/chat.ts]: I just prefer code execution to be explicit.

**Pull request merged**